addons/im_livechat/static/tests/chat_window_patch.test.js

```diff
@@ -72,6 +73,66 @@ test("closing a chat window with no message from admin side unpins it", async ()
     await contains(".o-mail-DiscussSidebarChannel", { count: 0, text: "Partner 2" });
 });
 
+test("Should open active livechat first on Tab key press", async () => {
+    const pyEnv = await startServer();
+    const guestIds = pyEnv["mail.guest"].create([
+        { name: "Visitor 1" },
+        { name: "Visitor 2" },
+        { name: "Visitor 3" },
+    ]);
+    const livechatChannelId = pyEnv["im_livechat.channel"].create({
+        name: "Test",
+        user_ids: [serverState.userId],
+    });
+    const channelIds = pyEnv["discuss.channel"].create(
+        guestIds.map((guestId) => ({
+            channel_type: "livechat",
+            channel_member_ids: [
+                Command.create({ partner_id: serverState.partnerId }),
+                Command.create({ guest_id: guestId }),
+            ],
+            livechat_active: true,
+            livechat_channel_id: livechatChannelId,
+            livechat_operator_id: serverState.partnerId,
+            create_uid: serverState.publicUserId,
+        }))
+    );
+
+    pyEnv["mail.message"].create([
+        {
+            author_guest_id: guestIds[0],
+            body: "livechat 1",
+            model: "discuss.channel",
+            res_id: channelIds[0],
+        },
+        {
+            author_guest_id: guestIds[1],
+            body: "livechat 2",
+            model: "discuss.channel",
+            res_id: channelIds[1],
+        },
+    ]);
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "Visitor 3" });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 3" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 3…']:focus");
+    // simulate visitor leaving
+    await withGuest(guestIds[0], () =>
+        rpc("/im_livechat/visitor_leave_session", { channel_id: channelIds[0] })
+    );
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 2 });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 2" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 2…']:focus");
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 3 });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 1" });
+    await contains("span", { text: "This livechat conversation has ended" });
+});
```

Comment:
Test doesn't look good enough to cover the case: it's possible the code simply order Tab from newest to oldest, therefore the ended livechat is at the end only because this is from the 1st visitor.

I would make visitor 2 leave instead, and I would post some messages in the different conversations in-between, just to give some robustness of the ordering. I would also increase livechat conversations to around 5, so that this is much more obvious that the ended livechats are at the end (and this looks more obviously consequence of actual code for this, rather than accidental like here).